### PR TITLE
chore(ci): enforce Node and Yarn versions

### PR DESCRIPTION
**Pull request by Nodonisko**
Everyone should install and use `nvm` to manage node versions or use Nix.

// edit 17.1.2022
Github CI will now use version from `.nvmrc`, fixed prettier formatting for `.yml` files, there is issue with Gitlab CI Docker image
// edit 19.1.2022
Fixed Docker image, added docs how to upgrade Node, ready to review

### CODE REVIEW

**Comment by tsusanka** [comment]
Note to nix fans: we have these versions already and [pinned](https://github.com/trezor/trezor-suite/blob/2d57aebd2e991028b55549d036f64cf20bb1031e/shell.nix#L4).

```
$ nix-shell
$ yarn --version                                                
1.22.17
$ node --version                 
v14.18.1
```

(Which I am not saying anything against this PR, we can have both. Nix lecture coming soon so we all can get on board.)

File: package.json [hunk 1]
```diff
@@ -5,8 +5,8 @@
     "repository": "https://github.com/trezor/trezor-suite.git",
     "license": "SEE LICENSE IN LICENSE.md",
     "engines": {
-        "node": ">=14",
-        "yarn": ">=1.22.0"
+        "node": "~14.18.1",
+        "yarn": "~1.22.17"
```
Comment (prusnak) [inline package.json]: If we are pinning to particular versions shouldn't this just be the following?

```suggestion
        "node": "14.18.1",
        "yarn": "1.22.17"
```

Comment (Nodonisko) [reply package.json]: I think that patch version doesn't matter that much, but we can pin exact version.

**Comment by Nodonisko** [comment]
> Note to nix fans: we have these versions already and [pinned](https://github.com/trezor/trezor-suite/blob/2d57aebd2e991028b55549d036f64cf20bb1031e/shell.nix#L4).
> 
> ```
> $ nix-shell
> $ yarn --version                                                
> 1.22.17
> $ node --version                 
> v14.18.1
> ```
> 
> (Which I am not saying anything against this PR, we can have both. Nix lecture coming soon so we all can get on board.)

Yea, if you using Nix then it doesn't matter, but it's common courtesy to have this for developers lazy to use Nix 😄 or for example some contributors from open source community

File: docs/misc/node_upgrade.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# How to upgrade node version?
+
+If you want to upgrade NodeJS version, you need to update it in two places:
```
Comment (vdovhanych) [inline docs/misc/node_upgrade.md:3]: I would also point out that we need to change the version in `shell.nix` based on the version that is in the channel we're pinning to.  So when someone updates the node version they remember to update it based on the version in nix.

File: docker/docker-compose.suite-ci.yml [hunk 3]
```diff
@@ -1,11 +1,11 @@
-version: "3"
+version: '3'
```
Comment (vdovhanych) [inline docker/docker-compose.suite-ci.yml]: why the change to single quotes? Is there any particular reason or only preferences?

**Comment by matejkriz** [comment]
Looks good, works as expected. 

We need to update [the main README](https://github.com/trezor/trezor-suite/blob/chore/nvm/README.md/#L16-L17 ) too 
So that newcomers can read that they need NVM and that there is an automation for the node version switching https://github.com/nvm-sh/nvm#deeper-shell-integration

File: docker/docker-compose.suite-ci.yml (hunk 3 above)
Comment (prusnak) [reply docker/docker-compose.suite-ci.yml]: I guess that is related to prettier reformat (`docker` folder has been removed from `.prettierignore` in this PR)

**Review by matejkriz — APPROVED** [review]

File: docs/misc/node_upgrade.md (hunk 2 above)
Comment (Nodonisko) [reply docs/misc/node_upgrade.md:3]: done

File: docker/docker-compose.suite-ci.yml (hunk 3 above)
Comment (Nodonisko) [reply docker/docker-compose.suite-ci.yml]: yea, I guess it was in ignore because prettier was broken for `yml` files (bad spacing size), now it is fixed so we should use it because of consistency

Comment (Nodonisko) [reply docker/docker-compose.suite-ci.yml]: overrided prettier config for `yml` files to use double quotes, which are more standart for `yml` I belive

**Review by vdovhanych — APPROVED** [review]
Lets do it

File: .gitlab-ci.yml [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
-image: registry.gitlab.com/satoshilabs/trezor/trezor-suite/base@sha256:32b21e049618d33737786d8b1c52d0d896f979ba305bc7453285209fdf3cdf92
+image: registry.gitlab.com/satoshilabs/trezor/trezor-suite/base@sha256:7002a4716e04b8b087dbd7b553c37724026949a0f0ee0c16618d5227d3224df0
 
 variables:
-  DEV_SERVER_URL: "https://suite.corp.sldev.cz"
-  STAGING_WALLET_SERVER_URL: "https://staging-wallet.trezor.io/wallet"
-  STAGING_SUITE_SERVER_URL: "https://staging-suite.trezor.io"
-  DESKTOP_APP_NAME: "Trezor-Suite"
-  GIT_CLEAN_FLAGS: "-ffdx -e .yarn"
+  DEV_SERVER_URL: 'https://suite.corp.sldev.cz'
```
Comment (matejkriz) [inline .gitlab-ci.yml]: apply prettier please

Comment (Nodonisko) [reply .gitlab-ci.yml]: done

**Review by vdovhanych — APPROVED** [review]
